src/chemistry/chemTabModel.cpp

```diff
@@ -62,6 +59,22 @@ ablate::chemistry::ChemTabModel::ChemTabModel(std::filesystem::path path) {
     inputFileStream.open(ipath.c_str(), std::ios::in);
     LoadBasisVectors(inputFileStream, speciesNames.size(), iWmat);
     inputFileStream.close();
+    //load source energy scaler
+    sourceEnergyScaler = (PetscReal *) malloc(2*sizeof(PetscReal));
+    inputFileStream.open(spath.c_str(), std::ios::in);
+    std::string line,value;
+
+    std::getline(inputFileStream, line);
+    std::stringstream lineStream1(line);
+    getline(lineStream1,value,' ');
+    sourceEnergyScaler[0] = std::stod(value);
+
+    std::getline(inputFileStream, line);
+    std::stringstream lineStream2(line);
+    getline(lineStream2,value,' ');
+    sourceEnergyScaler[1] = std::stod(value);
+
+    inputFileStream.close();
 }
 
 ablate::chemistry::ChemTabModel::~ChemTabModel() {
```

Comment:
Do we need to iterate through each component of Wmat and iWmat and free each component?

Reply:
Yes, I will add that.